x-pack/plugins/ml/public/application/memory_usage/memory_usage_page.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useState } from 'react';
+import { useTimefilter } from '@kbn/ml-date-picker';
+import { EuiFlexGroup, EuiFlexItem, EuiTabs, EuiTab } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { NodesList } from './nodes_overview';
+import { MlPageHeader } from '../components/page_header';
+import { JobMemoryTreeMap } from './memory_tree_map';
+import { useIsServerless } from '../contexts/kibana/use_is_serverless';
+
+enum TAB {
+  NODES,
+  MEMORY_USAGE,
+}
+
+export const MemoryUsagePage: FC = () => {
+  const serverless = useIsServerless();
+  const [selectedTab, setSelectedTab] = useState<TAB>(TAB.NODES);
+  useTimefilter({ timeRangeSelector: false, autoRefreshSelector: true });
+
+  return (
+    <>
+      <MlPageHeader>
```

Comment:
Does the memory usage for trained models only show up after synchronizing the saved objects? If so, this page needs the usual warning callout too I think.

Reply:
Added in https://github.com/elastic/kibana/pull/149419/commits/052561aed83bd228bc67d4c82e496fadfdaf1edd